.github/workflows/pythonpackage.yml

```diff
@@ -0,0 +1,96 @@
+name: Tests and lint
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - "v*"
+  pull_request:
+    branches:
+      - "*"
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        # Use a matrix for the python version here to make the rest of the lint job setup
+        # as similar as possible to the test job below.
+        python-version: [3.8]
+        lint-flags:
+          - "--run-only-fast-linters"
+          - "--run-only-pylint"
+          - "--run-only-mypy"
+          - "--run-only-bandit"
+          - "--run-only-sphinx-build"
+          - "--run-only-typing-copilot-tighten"
```

Comment:
Is splitting the different types of linting across a matrix for performance reasons, or so that the results are reported separately? If the former, is this really faster when accounting for setup time?

Reply:
`pylint` is the "long pole" in the entire test suite, at 5-6min on its own. `typing_copilot tighten` is around 4min. `bandit` and `mypy` are also 1-2 minutes each. `sphinx-build` is super fast but didn't feel like it made any sense to combine with the "fast linters" which are useful for local development (`flake8, black, isort` etc.).

So not all of them need to be split up — e.g. `mypy` and `bandit` could be combined without a net slowdown of the suite. But `pylint, mypy, typing_copilot, bandit` all do some pretty heavyweight AST munging that doesn't run particularly quickly, and at least to me there didn't seem to be a "conceptually clean" way to combine tools into fewer runs.

Reply:
Oh I missed that this is running `./scripts/lint.sh` which calls out to different services. I thought that the whole thing was one linter that could be configured to run different portions of itself.

Reply:
Yeah, we kind of bend over backwards here to make sure that the bot's lint checks are 100% reproducible locally. If we didn't care about that so much, in principle it would have been a lot easier to just use the GitHub Actions version of each of the linter tools when available, and have a CI-specific script for the rest. There are pros and cons both ways, and in this PR I tried to keep to the spirit of the old CI system as much as possible.

Reply:
So do we. `npm test` 🙃